### PR TITLE
682 fix user event trace incrementing

### LINKED ISSUE #682: Fix user event trace event incrementing
**Describe the bug**

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #682

Another shot at fixing the user event bug. The increments were not happening in the correct sequence.

### CODE REVIEW

File: src/vt/trace/trace.cc [hunk 1]
```diff
@@ -663,15 +662,11 @@ TraceEventIDType Trace::logEvent(LogType&& log) {
     break;
   case TraceConstantsType::UserEvent:
   case TraceConstantsType::UserEventPair:
-    if (not log.user_data().user_start) {
-      cur_event_++;
-    }
-    log.event = cur_event_;
+    log.event = log.user_data().user_start ? cur_event_ : cur_event_++;
```
Comment (PhilMiller) [inline src/vt/trace/trace.cc]: I'd favor sticking with separate read and increment statements for each of these instances, but especially this one. A ternary with side effects is going to be easily mistaken later.

Comment (lifflander) [reply src/vt/trace/trace.cc]: Ok, fixed.

File: src/vt/trace/trace.cc [hunk 2]
```diff
@@ -663,15 +662,17 @@ TraceEventIDType Trace::logEvent(LogType&& log) {
...
-    if (not log.user_data().user_start) {
-      cur_event_++;
+    TraceEventIDType event = no_trace_event;
+    if (log.user_data().user_start) {
+      event = cur_event_;
+    } else {
+      event = cur_event_++;
     }
-    log.event = cur_event_;
+    log.event = event;
```
Comment (nlslatt) [inline src/vt/trace/trace.cc]: I think you're overcomplicating this.  `event = cur_event_` and `event = cur_event_++` give `event` exactly the same value.

Comment (lifflander) [reply src/vt/trace/trace.cc]: Agreed, I'm just splitting out the branch that increments to make it clear, per @PhilMiller comment:
> I'd favor sticking with separate read and increment statements for each of these instances, but especially this one. A ternary with side effects is going to be easily mistaken later.

Comment (nlslatt) [reply src/vt/trace/trace.cc]: Why not make it
```
  log.event = cur_event_;
  if (not log.user_data().user_start) {
    cur_event_++;
  }
```

Comment (lifflander) [reply src/vt/trace/trace.cc]: Ok, done

**Pull request merged**